### ISSUE TITLE
User Interface for managed files

### ISSUE THREAD

**Issue by qrooel** [twpayne/chezmoi#4099]
## Feature request: user Interface for managed files

Thank you for all your work, chezmoi is absolutely great piece of software! :heart: 

I would like to request a feature. 
Nowadays I see more and more UIs in terminal apps.

Would it be possible to add one to the chezmoi? 
Similar to [vim-startify](https://github.com/mhinz/vim-startify), so we can see all managed files from a nice UI that we could traverse (and maybe even filter via [fzf](https://github.com/junegunn/fzf)) using VIM `hjkl` keys? :100: 

![image](https://github.com/user-attachments/assets/484feec6-8b8d-4e49-a22a-ae69dc9cf1fb)

**Comment by halostatue** [comment]
I don't want to speak for other contributors, but this does not seem likely to happen. Chezmoi itself is fairly complex and adding a TUI on top of that would increase the complexity fairly substantially, especially since it should *probably* be something a little closer to a ranger-like or midnight commander-like file manager.

I believe that someone could build an interface to do this by interpreting output in much the same way that `lazygit` or `lazydocker` does (I don't believe that either one uses a builtin library for interactions, but I could be wrong).

- `chezmoi data` provides a JSON formatted representation of runtime and `~/.config/chezmoi/chezmoi.$FORMAT` data, which could be used to drive all other command-line interactions.
- `chezmoi managed --format json --path-style all` provides a list of all managed files in both absolute and relative formats as well as the absolute target path.
- `chezmoi unmanaged --path-style absolute` provides a (partial) list of all unmanaged files (it terminates at the first unmanaged *directory* when traversing depth). It isn't JSON, but it is one file per line.

There may be a few other things that could be provided from the chezmoi side for parsing, but I don't think that it would be too difficult for someone to get a start on it. If there were specific issues someone faced while building such, I think there would be openness for adding interoperability, but we'd need to see what those were.

At least initially, I do not believe that there would be openness for moving code from `internal/` to a public interface for incorporation into a downstream program.

**Comment by KapJI** [comment]
Yeah, I think this is out of scope for chezmoi. There is a potential for another tool with more graphical UI which is built on top of chezmoi.

**Comment by qrooel** [comment]
Yep, that is true, thanks for response.

**Comment by twpayne** [comment]
> Similar to [vim-startify](https://github.com/mhinz/vim-startify), so we can see all managed files from a nice UI that we could traverse (and maybe even filter via [fzf](https://github.com/junegunn/fzf)) using VIM `hjkl` keys? 💯

You can do this with

```console
$ chezmoi managed | fzf
```

As others have said, a text UI is out of scope for chezmoi itself. chezmoi does have some limited terminal input for prompting the user occasionally, but a full user interface is a significantly larger and more opinionated project. chezmoi is designed to be composed with other programs, and a text UI could [build on top of chezmoi](https://www.chezmoi.io/developer-guide/building-on-top-of-chezmoi/).

So, this will not be worked on, but feel free to build on top of chezmoi!
